tests/system_tests_http1_adaptor.py

```diff
@@ -1262,5 +1287,192 @@ def test_multipart_aggregation(self):
+    """Verify the event-channel functionality."""
+    @classmethod
+    def setUpClass(cls):
+        super(Http1AdaptorEventChannelTest, cls).setUpClass()
+
+        cls.server_port = cls.tester.get_port()
+        cls.listener_port = cls.tester.get_port()
+
+        config = [
+            ('router', {'mode': 'interior',
+                        'id': 'HTTPEventChannel'}),
+            ('listener', {'role': 'normal',
+                          'port': cls.tester.get_port()}),
+
+            ('httpConnector', {
+                'name': 'httpConnector1',
+                'port': cls.server_port,
+                'host': '127.0.0.1',
+                'protocolVersion': 'HTTP1',
+                'eventChannel': 'true',
+                'address': 'closest/EventChannel'}),
+            ('httpListener', {
+                'name': 'httpListener1',
+                'port': cls.listener_port,
+                'host': '0.0.0.0',
+                'protocolVersion': 'HTTP1',
+                'eventChannel': 'true',
+                'address': 'closest/EventChannel'}),
+
+            ('address', {'prefix': 'closest',   'distribution': 'closest'}),
+            ('address', {'prefix': 'multicast', 'distribution': 'multicast'}),
+        ]
+        config = Qdrouterd.Config(config)
+        cls.router = cls.tester.qdrouterd('HTTP1EventChannel', config, wait=False)
+        # do not wait for connectors - servers haven't been created yet
+        cls.router.wait_ports()
+        cls.router.wait_startup_message()
+
+    @unittest.skipIf(not curl_available(), "test required 'curl' command not found")
+    def test_event_channel_curl(self):
+        """Verify event channel post via curl"""
+        TESTS = {
+            "POST": [
+                (RequestMsg("POST", "/POST/EventChannel/1",
+                            headers={"Content-Length": 33},
+                            body=b'This is not used dummy test entry'),
+                 ResponseMsg(204, reason="No Content",
+                             headers={"Content-Length": 0,
+                                      "Dummy-header": "OOGA BOOGA!"}),
+                 None),
+            ]
+        }
+
+        # Cannot use a context for the servers since this is POST event channel
+        # which does not return a response. Exiting the context will cause
+        # a POST /SHUTDOWN to occur, but the test will fail waiting for a
+        # response to the shutdown command
+
+        try:
+            server = TestServer.new_server(self.server_port,
+                                           self.listener_port,
+                                           TESTS)
+        except Exception as exc:
+            print(f"Failed to start test server: {exc}", flush=True)
+            raise
+
+        try:
+            self.router.wait_address('closest/EventChannel', subscribers=1)
+
+            # wait for the listener to activate
+            mgmt = self.router.qd_manager
```

Comment:
Perhaps this check for operStatus can be replaced with a call to `wait_http_listeners_up()` (https://github.com/skupperproject/skupper-router/blob/main/tests/system_tests_http1_adaptor.py#L334)

`wait_http_listeners_up()` also seems to check for operStatus

Reply:
Yes - good idea: added.